content/blog/kaggle_on_kubeflow.md

```diff
@@ -0,0 +1,105 @@
+- TGS Salt Identification Challenge (Segment salt deposits beneath the Earth's surface): $100,000
+
+For new data scientists, there are competitions that are interesting thought experiments. The most notable of these is predicting which persons would survive the Titanic disaster, which in fact we will use for our example below.  
+
+The Kaggle development platform itself is organized into competitions, scripts called "kernels", and datasets which are used to derive the models submitted to the competitions. There are also short form on-line classes for introductions to Python and machine learning.
+
+## Images
+### Kubeflow Notebooks 
+The Kubeflow project early on made a decision to curate, build, and maintain Jupyter notebook images with specific content in them. Since TensorFlow ranks highest in popularity among ML frameworks, it was an obvious choice to have that anchor the notebooks starting with TF 1.4.1. After that there is Keras, Sci-Kit Learn, pandas, and various libraries for JupyterHub widgets and so on. But we had to draw a line somewhere: the more content and layers that go into a Dockerfile, the more the derived image grows in size. Also, we wanted to support both CPU and GPU flavors of TF.
+
+Kubeflow opted for an opinionated balance between images that had somewhat manageable sizes (3-4GB) and core content, as opposed to installing all of the many different frameworks that are available. However, we modified ownership within the images to allow users to install any frameworks or libraries they wanted. We didn't want the initial experience for new developers using Kubeflow to be impacted by long download latency during docker pulls from the Kubeflow `gcr.io/kubeflow-images-public` container registry.
+
+### Kaggle Notebook
+Kaggle maintains [its own Python Docker image project](https://github.com/Kaggle/docker-python) which is used as the basis for Kubeflow to provide an image that has all the rich goodness of virtually every available Python ML framework and tool while also having the necessary mods for it to be easily deployed into a Kubeflow environment. But, as I mentioned previously, there is some cost associated with that.
+
+- it's a very large notebook, over 21 GB in size. Docker pulls and notebook launches can take a lengthy period of time
+- the versions of TensorFlow, PyTorch, XGBoost, and the other libraries included may change at any time
+- the base image size for Docker devicemapper is 10 GB, which won't be large enough to run this image, so look to alternate Docker storage drivers
+- the Kaggle image includes TensorFlow 1.9 or greater built with [AVX2 support](https://en.wikipedia.org/wiki/Advanced_Vector_Extensions#Advanced_Vector_Extensions_2), so the image may not run on some older CPU
+- unlike the Kubeflow curated notebooks, the default notebook user (`jovyan`) does not have the permissions to install new packages to global locations (but more on that below)
+
+### What's this all about?

```

Comment:
I'd move this section ("What's this all about?") up above the "Images" section, and change it to a level 2 heading. It's super important that people know the goal of this blog post, and this section tells it nicely. When I first read the "Images" section, I wondered why you were telling me all that. Then I read the "What's this all about?" section, and things became much clearer.